### ISSUE TITLE
seport configuration before selinux reboot

### ISSUE THREAD

**Issue by ryanerwin** [ansible/ansible-modules-extras#2077]
This issue has not previously been reported on GitHub
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

ansible-modules-extras/blob/devel/system/seport.py
##### ANSIBLE VERSION

```
ansible 2.0.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

Ansible Host: CentOS Linux release 7.2.1511 (Core)
Ansible Target: Red Hat Enterprise Linux Server release 5.6 (Tikanga)
##### SUMMARY

When a new system comes online, there's a long list of things to do, but a critical point is:
1. ensure that SElinux is turned on
2. use `seport` to open custom SSH port under SElinux
3. update SSH configuration for custom port
4. reboot server if necessary for SElinux configuration

Problem is you can't execute this with ansible seport as currently written because seport will fail to add a port to the whitelist if you just executed `selinux: policy=targeted state=enforcing` because the system must be rebooted first...

Yet if you haven't updated the whitelist, you'll never be able to reconnect after the reboot.
##### STEPS TO REPRODUCE

If you run this on a system that doesn't yet have SElinux enabled, it will fail with error: SELinux is disabled on this host.

```
  - name: prep for SElinux dependancies
    yum: state=present name={{ item }}
    with_items:
     - libselinux-python
     - policycoreutils-python

  - name: enable SELinux
    selinux: policy=targeted state=enforcing

  - name: make sure we've got YOUR PORT for SSH open via SELinux
    seport: ports=YOUR PORT proto=tcp setype=ssh_port_t state=present

  - name: standard SSH configuration
    lineinfile:
      dest: /etc/ssh/sshd_config
      state: present
      create: yes
      regexp: "{{ item.regexp }}"
      line: "{{ item.line }}"
      # validate: '/usr/sbin/sshd -t %s'
    with_items:
      - { regexp: '^#?Port\s', line: 'Port YOURPORT' }
      - { regexp: '^#?PermitRootLogin\s', line: 'PermitRootLogin no' }
      - { regexp: '^#?X11Forwarding\s', line: 'X11Forwarding no' }
    notify: restart ssh
```

##### EXPECTED RESULTS

Personally I think the command should work just fine on a system that has SElinux "enabled" via the configuration options, even if the actual SElinux process isn't yet engaged (via reading the configuration at reboot).

Note that even when I commented out the selinux check and tried to run it on the target host, I got:
    # if not selinux.is_selinux_enabled():
    #   module.fail_json(msg="SELinux is disabled on this host.")

`sudo python ./tmp/ansible-tmp-1461064894.01-221154378874892/seport
Traceback (most recent call last):
  File "./tmp/ansible-tmp-1461064894.01-221154378874892/seport", line 2220, in ?
    main()
  File "./tmp/ansible-tmp-1461064894.01-221154378874892/seport", line 252, in main
    result['changed'] = semanage_port_add(module, ports, proto, setype, do_reload)
  File "./tmp/ansible-tmp-1461064894.01-221154378874892/seport", line 137, in semanage_port_add
    seport.set_reload(do_reload)
AttributeError: portRecords instance has no attribute 'set_reload'`

Yet when I run the command directly in the shell with /usr/sbin/semanage port everything works properly.
##### ACTUAL RESULTS

```
fatal: [SOMEHOST]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"ports": "YOUR PORT", "proto": "tcp", "reload": true, "setype": "ssh_port_t", "state": "present"}, "module_name": "seport"}, "msg": "SELinux is disabled on this host."}
```

**Comment by chouseknecht** [comment]
@wimnat

**Comment by ansibot** [comment]
@dankeder, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dankeder** [comment]
Hi, thanks for the bug report. The thing is that SELinux policy can't be changed when SELinux is in disabled state.

It's possible to work-around this by setting SELinux to `permissive` mode first, reboot, set-up the port contexts etc., and only then switch to enforcing mode. This is what I ended up doing.

**Comment by dankeder** [comment]
(Adding "wontfix", as I don't see much I can do about this.)
